dbms/src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -1470,6 +1471,42 @@ bool ParserFunctionWithKeyValueArguments::parseImpl(Pos & pos, ASTPtr & node, Ex
     return true;
 }
 
+bool ParserTTLElement::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserKeyword s_to_disk("TO DISK");
+    ParserKeyword s_to_volume("TO VOLUME");
+    ParserKeyword s_delete("DELETE");
+    ParserStringLiteral parser_string_literal;
+    ParserExpression parser_exp;
+
+    ASTPtr expr_elem;
+    if (!parser_exp.parse(pos, expr_elem, expected))
+        return false;
+
+    ASTTTLElement::DestinationType destination_type = ASTTTLElement::DestinationType::DELETE;
+    String destination_name;
+    if (s_to_disk.ignore(pos)) {
+        destination_type = ASTTTLElement::DestinationType::DISK;
+    } else if (s_to_volume.ignore(pos)) {
+        destination_type = ASTTTLElement::DestinationType::VOLUME;
+    } else {
+        s_delete.ignore(pos);
```

Comment:
What if it will return `false`? Need check here.

Reply:
That will be fine, in this case this is a fall-back to previous syntax (without `DELETE`).

Reply:
That will be fine, in this case this is a fall-back to previous syntax (without `DELETE`).